### PR TITLE
Feat/remove package UI

### PR DESCRIPTION
**Pull request by bilaldbank**
This PR adds a minimal UI (built with Flutter web) for removing packages from our unpub deployment

### CODE REVIEW

File: unpublish/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: unpublish
```
Comment (bilaldbank) [inline unpublish/pubspec.yaml:1]: 1. The separate Flutter web project for displaying a UI for package removal from unpub

File: unpub/tool/pre_unpublish.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import 'dart:io';
```
Comment (bilaldbank) [inline unpub/tool/pre_unpublish.dart:1]: 2. This builds the flutter web project and copies the build files into unpub project as static strings.
```
--pwa-strategy=none to remove service worker and keep the build size small
--web-renderer=html to remove canvaskit and keep the build size small
--base-href=/delete/ as unpub_web is served at / so we need a subpath for serving the flutter app
```

File: unpub/lib/src/static/index.html.dart [hunk 3]
```diff
@@ -1696,11 +1696,35 @@ const content = """<!DOCTYPE html>
...
+
+        .banner-link {
+            text-decoration: none;
+        }
     </style>
     <link rel="icon" type="image/png" href="https://pub.dev/favicon.ico" />
     <script defer src="main.dart.js"></script>
   </head>
   <body>
+        <div class="banner">
```
Comment (bilaldbank) [inline unpub/lib/src/static/index.html.dart:1722]: 3. Add an HTML banner on everypage of unpub so it's easy to find the Flutter web app from main unpub UI

File: unpub/lib/src/mongo_store.dart [hunk 4]
```diff
@@ -73,16 +73,26 @@ class MongoStore extends MetaStore {
 
   @override
   Future<UnpubQueryResult> queryPackages({
-    required size,
-    required page,
-    required sort,
+    size,
```
Comment (bilaldbank) [inline unpub/lib/src/mongo_store.dart:76]: 4. size, page, and sort are made optional to support the new endpoint for getting all packages

File: unpub/lib/src/app.dart [hunk 5]
```diff
@@ -142,6 +146,25 @@ class App {
 
   Router get router => _$AppRouter(this);
 
+  @Route.get('/api/packages')
+  Future<shelf.Response> getPackageNames(shelf.Request req) async {
```
Comment (bilaldbank) [inline unpub/lib/src/app.dart:150]: New endpoint for returning list of all packages

File: unpub/lib/src/app.dart [hunk 6]
```diff
@@ -545,6 +567,26 @@ class App {
         headers: {HttpHeaders.contentTypeHeader: ContentType.html.mimeType});
   }
 
+  /// Return Flutter web app
+  @Route.get('/delete')
+  @Route.get('/delete/')
+  Future<shelf.Response> flutterHtml(shelf.Request req) async {
```
Comment (bilaldbank) [inline unpub/lib/src/app.dart:573]: endpoint for Flutter web app

**Comment by bilaldbank** [comment]
https://github.com/din-global/unpub/assets/119291964/afb052e9-6336-4030-8424-969d25d45894

**Review by tasnim-din — APPROVED** [review]
As we are not expecting many changes in this part of the code, we don't need to add tests. Approving and merging this for @ydmk to deploy when it is possible.

**Pull request merged**